wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMenuItemRenderer.java

```diff
@@ -1,20 +1,57 @@
 package com.github.bordertech.wcomponents.render.webxml;
 
+import com.github.bordertech.wcomponents.MenuSelectContainer;
 import com.github.bordertech.wcomponents.WComponent;
 import com.github.bordertech.wcomponents.WMenuItem;
+import com.github.bordertech.wcomponents.WebUtilities;
 import com.github.bordertech.wcomponents.XmlStringBuilder;
 import com.github.bordertech.wcomponents.servlet.WebXmlRenderContext;
-import com.github.bordertech.wcomponents.util.Util;
 
 /**
  * The Renderer for {@link WMenuItem}.
  *
  * @author Yiannis Paschalidis
+ * @author Mark Reeves
  * @since 1.0.0
  */
 final class WMenuItemRenderer extends AbstractWebXmlRenderer {
 
 	/**
+	 * A WMenuItem is selectable if it is explicitly set as selectable (or not selectable) or if its own selectability is not set if its nearest
+	 * neighbour selection container is selectable.
+	 * @param item the WMenuItem to test
+	 * @return {@code true} if the WMenuItem is selectable
+	 */
+	private boolean isSelectable(final WMenuItem item) {

```

Comment:
Do you think it is better for this logic to be in WMenuItem.

There is already a "isSelectable" on WMenuItem which is deprecated and just returns "getSelectablility". 

What if the depraction was removed and repurpose the method to include this logic and remove setSelectable.

Might be good to also put the "getRole" logic on WMenuItem.